### PR TITLE
[P4-968] Prison risk logic

### PR DESCRIPTION
**Pull request by teneightfive**
This creates logic to split how different journeys will see the release status question.

Users sending some from Police/STC/SCH will see it asked as part of the risk step.

Prison users will see it asked as a separate step.

### From Police

image

### From Prison

image

### CODE REVIEW

File: app/move/controllers/create/assessment.js [hunk 1]
```diff
@@ -12,18 +12,26 @@ class AssessmentController extends CreateBaseController {
...
 
-      req.form.options.fields[assessmentCategory] = implicitField
+        req.form.options.fields[assessmentCategory] = {
+          ...implicitField,
+          items: implicitQuestions
+            .map(fieldHelpers.mapAssessmentQuestionToConditionalField)
+            .map(fieldHelpers.mapAssessmentQuestionToTranslation)
+            .map(fieldHelpers.mapReferenceDataToOption),
+        }
+      }
```
Comment (barbaracassani) [inline app/move/controllers/create/assessment.js]: Maybe this block of code that populates `assessmentCategory` should go into its own method -- or possibly just the block that builds `items`.

File: app/move/fields/create.js [hunk 2]
```diff
@@ -325,6 +328,10 @@ module.exports = {
   self_harm: assessmentQuestionComments,
   concealed_items: assessmentQuestionComments,
   other_risks: requiredAssessmentQuestionComments,
+  not_for_release: {
+    ...requiredAssessmentQuestionComments,
+    explicit: true,
```
Comment (barbaracassani) [inline app/move/fields/create.js]: Glad it gets reused... :-)

File: app/move/controllers/create/assessment.js (hunk 1 above)
Comment (teneightfive) [reply app/move/controllers/create/assessment.js]: I was kind of thinking of refactoring but didn't want to go down a rabbit hole just yet after we spent a fair bit of time on it last time. 

The main aim was to ensure it didn't create the implicit field if there were no implicit questions for that step.

Comment (teneightfive) [reply app/move/controllers/create/assessment.js]: I did start a refactor to try and remove the need for the `__yesno` suffix being required between a few files so I think we can revisit that later.

Comment (barbaracassani) [reply app/move/controllers/create/assessment.js]: 👍

**Pull request merged**